### PR TITLE
Various fixes for conmon

### PR DESCRIPTION
**Pull request by alexlarsson**
I've been reviewing the conmon code, and here is a bunch of fixes for issues that I found.
The comments on each commit describes the issues pretty well, and I don't think any of these are controversial, so I made it a single pull request.

### CODE REVIEW

**Comment by runcom** [comment]
Thanks Alex, you need to sign the cncf cla and re-push (force) to re-kick the bot.

File: conmon/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@ src = $(wildcard *.c)
 obj = $(src:.c=.o)
 
 override LIBS += $(shell pkg-config --libs glib-2.0 json-glib-1.0)
-override CFLAGS += -std=c99 -Wall -Wextra $(shell pkg-config --cflags glib-2.0 json-glib-1.0)
+override CFLAGS += -std=c99 -Wall -Wextra $(shell pkg-config --cflags glib-2.0)
```
Comment (runcom) [inline conmon/Makefile]: There are currently some entries for the json-glib deps that can be removed also:
```
contrib/test/crio-integration-playbook.yaml
47:      - json-glib-devel
74:      - json-glib-devel

conmon/Makefile
4:override LIBS += $(shell pkg-config --libs glib-2.0 json-glib-1.0)
5:override CFLAGS += -std=c99 -Wall -Wextra $(shell pkg-config --cflags glib-2.0 json-glib-1.0)

conmon/conmon.c
21:#include <json-glib/json-glib.h>

Dockerfile
22:    libjson-glib-dev \

README.md
64:  json-glib-devel \
82:  libjson-glib-dev \
```
Absolutely not blocking, can be cleaned in a follow up.

**Comment by mrunalp** [comment]
The changes look great. Thanks! We need to wait for travis and node e2e tests to come back.

**Comment by alexlarsson** [comment]
Yay, passed

**Comment by runcom** [comment]
LGTM

**Comment by mrunalp** [comment]
Just noticed while testing locally that we are missing #include <sys/uio.h> for writev. (We can fix that in a follow-on).

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**